### ISSUE TITLE
bestChannelAtStation in waveformVectorArm incorrectly fails some channels 

### ISSUE THREAD

**Issue by seisman** [crotwell/sod#3]
Following recipe correctly retrieves three waveforms of `IU.CCM.00.BH?`:
```
<?xml version="1.0"?>
<sod>
    <eventArm>
        <fdsnEvent>
            <originTimeRange>
                <startTime>2018-06-01T00:00:00.000Z</startTime>
                <endTime>2018-07-01T00:00:00.000Z</endTime>
            </originTimeRange>
            <magnitudeRange>
                <min>6.0</min>
                <max>7.0</max>
            </magnitudeRange>
            <originDepthRange>
                <unit>KILOMETER</unit>
                <min>0</min>
                <max>75</max>
            </originDepthRange>
		</fdsnEvent>
        <printlineEventProcess/>
    </eventArm>

    <networkArm>
        <fdsnStation>
            <networkCode>IU</networkCode>
            <stationCode>CCM</stationCode>
		    <channelCode>BH*</channelCode>
        </fdsnStation>
        <printlineChannelProcess/>
    </networkArm>

    <waveformArm>
        <bestChannelAtStation/>

        <phaseRequest>
            <model>iasp91</model>
            <beginPhase>ttp</beginPhase>
            <beginOffset>
                <unit>SECOND</unit>
                <value>-600</value>
            </beginOffset>
            <endPhase>tts</endPhase>
            <endOffset>
            <unit>SECOND</unit>
            <value>300</value>
            </endOffset>
        </phaseRequest>

        <fdsnDataSelect/>

        <sacWriter/>
        <printlineSeismogramProcess/>
    </waveformArm>
</sod>
```

However, if I change `waveformArm` to `waveformVectorArm`, the recipe would get no seismograms. Fail.log says all channels are failing due to bestChannelAtStation subsetter:
```
2018-10-23 15:20:27,940 INFO WaveformArm 2 Fail.WaveformVector - ECGroup: Event: Unknown | 06/21/2018 21:13:32 GMT | Mag: 6.1 | Depth 28000.00 m | (-17.7905, 168.0568) IU.1988-01-01T00:00:00.000Z.CCM.00.BHZ.2018-06-09T18:00:00.000Z , IU.1988-01-01T00:00:00.000Z.CCM.10.BH1.2018-06-09T18:00:00.000Z , IU.1988-01-01T00:00:00.000Z.CCM.10.BH2.2018-06-09T18:00:00.000Z ,  Event channel subsetter rejected: BestChannelAtStation:FAIL:
2018-10-23 15:20:27,952 INFO WaveformArm 2 Fail.WaveformVector - ECGroup: Event: Unknown | 06/21/2018 21:13:32 GMT | Mag: 6.1 | Depth 28000.00 m | (-17.7905, 168.0568) IU.1988-01-01T00:00:00.000Z.CCM.10.BHZ.2018-06-09T18:00:00.000Z , IU.1988-01-01T00:00:00.000Z.CCM.00.BH1.2018-06-09T18:00:00.000Z , IU.1988-01-01T00:00:00.000Z.CCM.00.BH2.2018-06-09T18:00:00.000Z ,  Event channel subsetter rejected: BestChannelAtStation:FAIL:
2018-10-23 15:20:28,423 INFO WaveformArm 1 Fail.WaveformVector - ECGroup: Event: Unknown | 06/21/2018 21:13:32 GMT | Mag: 6.1 | Depth 28000.00 m | (-17.7905, 168.0568) IU.1988-01-01T00:00:00.000Z.CCM.00.BHZ.2018-06-09T18:00:00.000Z , IU.1988-01-01T00:00:00.000Z.CCM.10.BH1.2018-06-09T18:00:00.000Z , IU.1988-01-01T00:00:00.000Z.CCM.10.BH2.2018-06-09T18:00:00.000Z ,  Event channel subsetter rejected: BestChannelAtStation:FAIL:
2018-10-23 15:20:28,431 INFO WaveformArm 1 Fail.WaveformVector - ECGroup: Event: Unknown | 06/21/2018 21:13:32 GMT | Mag: 6.1 | Depth 28000.00 m | (-17.7905, 168.0568) IU.1988-01-01T00:00:00.000Z.CCM.10.BHZ.2018-06-09T18:00:00.000Z , IU.1988-01-01T00:00:00.000Z.CCM.00.BH1.2018-06-09T18:00:00.000Z , IU.1988-01-01T00:00:00.000Z.CCM.00.BH2.2018-06-09T18:00:00.000Z ,  Event channel subsetter rejected: BestChannelAtStation:FAIL:
```

**Comment by crotwell** [comment]
I think SOD is having trouble grouping the channels into 3 components. Definitely a bug. I believe I have fixed it in the new version (4.0) that I have been working on, but it is not yet ready for release, probably months or more before ready.

As a workaround in this case, if you add
```
        <channelOR>
           <siteCode></siteCode>
           <siteCode>00</siteCode>
        </channelOR>
```
to the recipe just below the `</fdsnStation>` line, it will eliminate the 10 and 60 sites (aka loc ids) and then SOD will correctly group the channels and get seismograms. In most cases this is ok as the convention, for the most part, in seismology is to have the "main" channels be either space-space or 00 for the loc id.

**Comment by seisman** [comment]
@crotwell Thanks for your reply. Looking forward to the next release.
